volatility3/framework/layers/registry.py

```diff
@@ -66,7 +66,7 @@ def __init__(
         # Win10 17063 introduced the Registry process to map most hives.  Check
         # if it exists and update RegistryHive._base_layer
         for proc in pslist.PsList.list_processes(
-            self.context, self.config["base_layer"], self.config["nt_symbols"]
+            context=self.context, kernel_module_name=self.config["kernel_module_name"]
```

Comment:
Just want to get a little clarity on this, where does `kernel_module_name` get defined as the parameter this takes?  I would've expected a `requirements` method or something, but it doesn't seem to have one and that makes me worry that it's just convention that we look for that particular config name.  It might be right, but it will need fixing (at the very least an issue so we don't forget to investigate it).

It and the dependency on pslist needs to go too.

Reply:
I added it to here as the code path through hivelist is what creates a `subconfig` that the registry layer uses:

https://github.com/volatilityfoundation/volatility3/blob/a553ee39212e189e6866044ab80ecb5190f1a1aa/volatility3/framework/plugins/windows/registry/hivelist.py#L181

This is also the code path that will need to be updated to satisfy the ticket we made about the registry layer walking the process list (it all boils down to the same root cause in hivelist.py):

https://github.com/volatilityfoundation/volatility3/issues/1645

So all the related hivelist code will be cleaned up in 1645 anyway.

Reply:
Ok, thanks, that was the bit that was missing for me.  5:)